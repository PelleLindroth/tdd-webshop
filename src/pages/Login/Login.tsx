import { useState } from 'react'
import { validateUser, IUser } from '../../mocks/users'
import { useNavigate } from 'react-router-dom'

interface LoginProps {
  setUser: React.Dispatch<React.SetStateAction<IUser | null>>
}

const Login = (props: LoginProps) => {
  const { setUser } = props

  const [error, setError] = useState<boolean>(false)
  const [email, setEmail] = useState<string>('')
  const [password, setPassword] = useState<string>('')

  const navigate = useNavigate()

  const handleLogin = (e: React.FormEvent) => {
    e.preventDefault()
    setError(false)

    if (email && password) {
      const user = validateUser(email, password)
      if (user) {
        setUser(user)
        navigate('/profile')
      } else {
        setError(true)
      }
    }
  }

  return (
    <>
      <h1>LOGIN</h1>
      <form name="login-form" onSubmit={handleLogin}>
        <label htmlFor="email">Email</label>
        <input name="email" id="email" type="text" value={email} onChange={e => setEmail(e.target.value)} />
        <label htmlFor="password">Password</label>
        <input
          name="password"
          id="password"
          type="text"
          value={password}
          onChange={e => {
            setPassword(e.target.value)
          }}
        />
        <button disabled={!email.length || !password.length}>Login</button>
      </form>
      {error && <p style={{ color: 'red' }}>Email and password does not match our records</p>}
    </>
  )
}

export default Login
