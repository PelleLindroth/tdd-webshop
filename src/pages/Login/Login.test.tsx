import { shallow } from 'enzyme'
import Login from './Login'

describe('Login', () => {
  it('renders Login component correctly', () => {
    const wrapper = shallow(<Login />)

    expect(wrapper).toMatchSnapshot()
  })
})
