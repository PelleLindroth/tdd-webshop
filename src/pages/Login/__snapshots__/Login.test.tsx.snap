// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders Login component correctly 1`] = `
<h1>
  LOGIN
</h1>
`;
